### PR TITLE
Added text details to the send state change notifications

### PR DESCRIPTION
**Pull request by monich**
Allows to forward operator's errors messages to the user. Since it changes the D-Bus protocol between `mms-engine` and `commhistoryd`, it requires https://github.com/nemomobile/commhistory-daemon/pull/78 in order to actually work.

### CODE REVIEW

File: mms-handler-dbus/spec/org.nemomobile.MmsHandler.xml [hunk 1]
```diff
@@ -225,6 +225,14 @@
...
       -->
       <arg direction="in" type="i" name="state"/>
+      <!--
+          Optional detail string. In most cases it's empty by in case
+          if the message was refused by the operator it contains the
+          value of X-Mms-Response-Text header from M-Send.conf PDU.
+          It could explain the reason for the failure and should be
+          displayed to the user.
+      -->
+      <arg direction="in" type="s" name="details"/>
```
Comment (thp) [inline mms-handler-dbus/spec/org.nemomobile.MmsHandler.xml]: This changes the D-Bus API of this middleware component -- are all consumers updated accordingly?

File: mms-handler-dbus/src/mms_handler_dbus.c [hunk 2]
```diff
@@ -377,7 +378,8 @@ mms_handler_dbus_message_send_state_changed(
         mms_handler_ref(handler);
         mms_handler_busy_inc(handler);
         org_nemomobile_mms_handler_call_message_send_state_changed(proxy, id,
-            state, NULL, mms_handler_dbus_call_done, handler);
+            state, details ? details : "", NULL, mms_handler_dbus_call_done,
```
Comment (thp) [inline mms-handler-dbus/src/mms_handler_dbus.c]: `details ? details : ""` can be rewritten as `details ?: ""`

**Comment by thp** [comment]
Rest LGTM

File: mms-handler-dbus/spec/org.nemomobile.MmsHandler.xml (hunk 1 above)
Comment (monich) [inline mms-handler-dbus/spec/org.nemomobile.MmsHandler.xml]: commhistoryd is the only one.

File: mms-handler-dbus/src/mms_handler_dbus.c (hunk 2 above)
Comment (monich) [inline mms-handler-dbus/src/mms_handler_dbus.c]: Wow, I didn't know. Probably a GNU extension.

**Pull request merged**